scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -850,42 +866,69 @@ def _sanitize_runners_file(self):
         """
-        runners_dir_path: str = os.path.join(self.instance.build_dir, 'zephyr')
-        runners_file_path: str = os.path.join(runners_dir_path, 'runners.yaml')
-        if not os.path.exists(runners_file_path):
-            return
+        runners_dir : str = os.path.join(self.instance.build_dir, 'zephyr')
+        runners_files: List[str] = [os.path.join(runners_dir, 'runners.yaml')]
 
-        with open(runners_file_path, 'rt') as file:
-            runners_content_text = file.read()
-            runners_content_yaml: dict = yaml.safe_load(runners_content_text)
+        if self.testsuite.sysbuild:
+            for subfolder in self.sub_folders:
+                sub_dir: str = os.path.join(self.instance.build_dir, subfolder, 'zephyr')
+                sub_runners_file: str = os.path.join(sub_dir, 'runners.yaml')
+                runners_files.append(sub_runners_file)
 
-        if 'config' not in runners_content_yaml:
-            return
+        for _runner in runners_files:
+            if not os.path.exists(_runner):
+                continue
 
-        runners_config: dict = runners_content_yaml['config']
-        binary_keys: List[str] = ['elf_file', 'hex_file', 'bin_file']
+            runners_content_yaml = {}
+            with open(_runner, 'rt') as file:
+                runners_content_text = file.read()
+                runners_content_yaml: dict = yaml.safe_load(runners_content_text)
 
-        for binary_key in binary_keys:
-            binary_path = runners_config.get(binary_key)
-            # sanitize only paths which exist and are absolute
-            if binary_path is None or not os.path.isabs(binary_path):
+            if 'config' not in runners_content_yaml:
                 continue
-            binary_path_relative = os.path.relpath(binary_path, start=runners_dir_path)
-            runners_content_text = runners_content_text.replace(binary_path, binary_path_relative)
 
-        with open(runners_file_path, 'wt') as file:
-            file.write(runners_content_text)
+            runners_config: dict = runners_content_yaml['config']
+            binary_keys: List[str] = ['elf_file', 'hex_file', 'bin_file']
+
+            for binary_key in binary_keys:
+                binary_path = runners_config.get(binary_key)
+                # sanitize only paths which exist and are absolute
+                if binary_path is None or not os.path.isabs(binary_path):
+                    continue
+                if self.testsuite.sysbuild:
+                    start_path = os.path.dirname(os.path.abspath(_runner))
+                else:
+                    start_path = os.path.dirname(os.path.dirname(os.path.abspath(_runner)))
+                bin_path_relative = os.path.relpath(binary_path, start=start_path)
+                runners_content_text = runners_content_text.replace(binary_path, bin_path_relative)
+
+            with open(_runner, 'wt') as file:
+                file.write(runners_content_text)
```

Comment:
oh my, this seems like `twister` is manipulating the files produced by the build system.

Twister should really not be messing with files created by the build system.
If twister wants to update such files before packaging for relocation purposes, then please do so in a dedicated folder in the build tree, and not by touching files controlled by the build system.

I'm not really keen on the possible risk of having to track down bugs because twister has changed content behind the back of the build system.

I notice this is not something new, but the knowledge about it has been unknown to me until reviewing this PR.

@nordicjm and @PerMac thoughts ?

Reply:
@tejlmand , I am very unacceptible on this, the former twister change method is accepted and I can not be accept by doing the sameway. so I think this is just to block me to change this. so I will resort to use other tools, not in sysbuild.

Reply:
excuse me ?

I did not block the PR over this, but when noticing potential issues in code should in my opinion always be highlighted and relevant maintainers be notified.

As I mention in the comment,
> I'm not keen on this

and
> I notice this is not something new

is not the same as comments given here, where the wording `not accepted / approved` is used:
https://github.com/zephyrproject-rtos/zephyr/pull/64551#discussion_r1404264176
https://github.com/zephyrproject-rtos/zephyr/pull/64551#discussion_r1404272056

or do @hakehuang mean that code with potential side effects should not be highlighted / relevant people notified when such code is noticed ?

Reply:
and @hakehuang the comment should at least be addressed, addressing could be in the form:
- Issue will be fixed immediately
- We accept for now, but as there haven't been reports of problems in practice, but will create follow-up enhancement to address in future / make visible to others that a potential problem exists.
- thanks for raising awareness, we accept the risk / annoyance but consider it low enough to ignore
- ....

also remember that in future when someone does a git blame then this PR will show up, and others will be able to find this comment and understand why code is like this / why it was accepted.

Reply:
> and @hakehuang the comment should at least be addressed, addressing could be in the form:
> 
> * Issue will be fixed immediately
> * We accept for now as there haven't been reports of problems in practice, but will create follow-up enhancement to address in future / make visible to others that a potential problem exists.
> * thanks for raising awareness, we accept the risk / annoyance but consider it low enough to ignore
> * ....
> 
> also remember that in future when someone does a git blame then this PR will show up, and others will be able to find this comment and understand why code is like this / why it was accepted.

@tejlmand , you are the code owner, and I do respect your judegement. And as this feature enhancement takes so much time for me. and I have to think about another approach for this.